src/main/java/com/questhelper/requirements/util/LogicType.java

```diff
@@ -33,27 +33,36 @@
 public enum LogicType
 {
 	/** Returns true only if all inputs match the supplied predicate. */
-	AND(Stream::allMatch),
+	AND(Stream::allMatch, (n1, n2) -> n1.doubleValue() == n2.doubleValue()),
 	/** Returns true if any inputs match the supplied predicate. */
-	OR(Stream::anyMatch),
+	OR(Stream::anyMatch, (n1, n2) -> n1.doubleValue() > 0.0D),
 	/** The output is false is all inputs match the supplied predicate. Otherwise returns true. */
-	NAND((s,p) -> !s.allMatch(p)),
+	NAND((s,p) -> !s.allMatch(p), (n1, n2) -> n1.doubleValue() < n2.doubleValue()),
 	/** Returns true if all elements do not match the supplied predicate. */
-	NOR(Stream::noneMatch),
+	NOR(Stream::noneMatch, (n1, n2) -> n1.doubleValue() == 0.0D),
 	/** Returns true if either, but not both, inputs match the given predicate.
 	 * This only tests the first two elements of the stream.
 	 */
-	XOR((s,p) -> s.filter(e -> p.test(e)).limit(2).count() == 1),
+	XOR((s,p) -> s.filter(p).limit(2).count() == 1, (n1, n2) -> {
+		return (n1.doubleValue() > 0.0D && !(n2.doubleValue() > 0.0D)) || (n2.doubleValue() > 0.0D && !(n1.doubleValue() > 0.0D));
+	}),
```

Comment:
The `XOR` logic is weird. Normally you don't XOR when comparing but I did the best I could.
The logic flow is 
```java
if  (first_num > 0 && !(second_num > 0)) || ((second_num > 0 && !(first_num > 0))
```
There's probably a nicer way to do it, but that's the only way that makes sense for XOR'ing two numbers